pkg/sql/parser/eval.go

```diff
@@ -1750,36 +1756,27 @@ func evalArrayCmp(
 		}
 		b := d.(*DBool)
 		res := *b != DBool(not)
-		if res {
-			anyTrue = true
-		} else {
-			allTrue = false
-		}
-	}
-
-	if all {
-		if !allTrue {
+		if res && any {
```

Comment:
I optimized this loop through each `Datum` in our slice of `Datum`s such that it short circuits/early terminates when:

1. `ANY`: result is encountered that results in `true` (formally, there exists a right such that the predicate evaluates to `true`, "any" is true, done)
2. `ALL`: result is encountered that results in `false` (there exists a `right` such that the predicate evaluates to `false`, not all are true, we are finished)

This results in statements such as this
```
SELECT 1 = ALL(SELECT * FROM generate_series(1,1000000))
```
to terminate after evaluating `1 = 2` instead of iterating until the end.